homeworks/solutions/torantie/2-ddg/README.md

```diff
@@ -0,0 +1,74 @@
+# Homework 2: Visualizing Curvatures
+
+In this homework, you will be working on the implementation of curvature
+visualization for manifold triangle meshes.
+
+## Skeleton
+
+The skeleton for this homework is structured as follows:
+
+```
+2-ddg
+├── package.json
+├── package-lock.json
+├── README.md          <--- You work on this file
+├── src
+│   ├── assets
+│   │   └── bunny_tri.obj
+│   ├── colors.js
+│   ├── halfedge.js    <--- You work on this file
+│   ├── main.js
+│   ├── renderer.js
+│   └── vector.js
+└── webpack.config.js
+```
+
+You coding tasks are located in the `halfedge.js`.
+Please look for `TODO:` in these two files and complete them to archive
+the reference results listed below.
+
+> It might be useful to read the code in `main.js`.
+
+You should also document your implementation process. Specifically, you
+should answer the following questions:
+
+1. Which code snippet (report in line numbers) in the `halfedge.js` is the most time consuming for you to implement? Explain your coding experience and encountered challenges briefly.
+
+Creating the halfedge structure was the most time consuming (lines 233-385). Specifically line 268-280 of the halfedge structure creation took the longest, since it was about creating and linking the halfedges. Overall errors in the halfedge structure resulted in a harder time implementing everything else. So often times i had to go back to the structure and look for errors, making it very time consuming.
```

Comment:
Constructing the halfedge structure is the most tricky part of the project, and by accomplishing it you received a lot of understanding on how to deal with mesh structure in a more advanced way :)